### DISCUSSION TITLE
User Identity: humanId versus id/displayName versus name

### DISCUSSION THREAD

**Discussion by LenaNoctua** [git-bug/git-bug#591] [Q&A] [answered]
Can anybody tell why there is a truncated form of the id and when it should be used for the user identity? And also when name should be used in contrast to displayName? I saw the comments in the graphql file, but they don't seem to be enough for me to fully understand. Thanks in advance

**Comment by MichaelMure** [comment, marked as answer]
Human IDs are just truncated (prefix of) IDs. It's the exact same concept as when you have short hash in git instead of the whole long thing. You can use a short prefix of the full thing, as long as there is no ambiguity (two or more object with the same prefix you provided). It's just a UX way to make those less cumbersome for humans. In a user interface, it really depend on the context to choose which one you want: more or less details.

In the identities, most (if not all) fields are optional. Name IIRC is optional. DisplayName is providing a string representing the identity, based on those other fields. It might be the name, it might be the login, it might be both, or something else. In a user interface, that's what you want to display, with possibly a tooltip or a semi-hidden way to show the other fields if that make sense.
